### DISCUSSION TITLE
Enhancements and New Features for Deskreen

### DISCUSSION THREAD

**Discussion by pavlobu** [pavlobu/deskreen#50] [Ideas]
Hi all. 👋  Here you can share your ideas on how to make Deskreen better. Leave your ideas about new features, improvements new translations (i18n) you'd like to see in future releases of Deskreen.

**IMPORTANT NOTE about this thread:**

> This is not a feature request channel. So far I'm the main contributor and I have my own agenda. I'm developing Deskreen in my
> spare time out of my main work duties. If you want some feature to appear in soonest release of Deskreen there are there are two options for you:
>
> 1. Become a Business Partner on [Opencollective](https://opencollective.com/deskreen) or [Patreon](https://www.patreon.com/deskreen) and have an exclusive right to request this feature. It will be then prioritized as important and made in next release if other security issues don't come up. Remarks are: the feature should not be an R&D related one (like add touch and drawing support for tablets or add driver support for some operating system); the feature will be created in my spare time with no crunches or rushes from anyone's side.
> 2. Contribute with code and cooperate with me. The most active code contributors will be the successors of this repo and will get
>    owner privileges for this repo. How to become an active contributor? - By implementing features or doing bug fixes. Just open an
>    issue and we will discuss what you can help with.

---

**Comment by aaylnx** [comment]
Is there any way to keep the same port number available for use after a device successfully connects and disconnects? That way I could just bookmark and quickly reconnect after use a first time rather than having to type the number in again. It would just speed things up a bit.

Also, might there be a way to give the user the option to not have to grant access on a device which has successfully connected before?  That would also speed things up in terms of disconnecting and reconnecting.

If these options are already there, I am not seeing them.

**Reply by pavlobu** [reply]
I call it "autoconnect trusted devices" feature, Its not there and it will be in some of future releases. I think it may be coming with code signing before or after. I know it is important. I also want it to appear in deskreen.

note everything that comes after / in adress bar is not a port, it is a websocket room number in our case.

Cheers,
Paul

**Reply by aaylnx** [reply]
Great! 

Yes, I knew that the numbers after the slash were not a port number, but I've not had any experience dealing with websockets. Good to know!  Good to know there is planning for this feature to improve an already excellent app!

**Reply by imneonizer** [reply]
> I call it "autoconnect trusted devices" feature, Its not there and it will be in some of future releases. I think it may be coming with code signing before or after. I know it is important. I also want it to appear in deskreen.
> 
> note everything that comes after / in adress bar is not a port, it is a websocket room number in our case.
> 
> Cheers,
> Paul

Check out snapdrop.net they automatically scan websocket room number

**Reply by pavlobu** [reply]
> > I call it "autoconnect trusted devices" feature, Its not there and it will be in some of future releases. I think it may be coming with code signing before or after. I know it is important. I also want it to appear in deskreen.
> > note everything that comes after / in adress bar is not a port, it is a websocket room number in our case.
> > Cheers,
> > Paul
> 
> Check out snapdrop.net they automatically scan websocket room number

great, you can submit a pull request with it

**Reply by pavlobu** [reply]
> > I call it "autoconnect trusted devices" feature, Its not there and it will be in some of future releases. I think it may be coming with code signing before or after. I know it is important. I also want it to appear in deskreen.
> > note everything that comes after / in adress bar is not a port, it is a websocket room number in our case.
> > Cheers,
> > Paul
> 
> Check out snapdrop.net they automatically scan websocket room number

please note that Deskreen is designed NOT to use www, it only works on local networks for better security

**Reply by beloware** [reply]
Exactly this feature i miss badly or simply turn off the pairing security step. 
Which is greyed out at least my case.
This makes a bit inconvinient to use at laggy dropping whatever networks.

**Reply by EDIflyer** [reply]
Ooh this would be a great addition - esp for reconnecting to remote devices over VPN, that would save having to open up remote access to create a new session, scan QR code, etc. Perhaps just an option to 'allow reconnects for this device' when approving access in the first place? You could always then revoke access from the manage connections screen.  Perhaps based on some combo of incoming MAC address + websocket ID? (obviously MAC could be faked but would need to know corresponding websocket and also given it only runs on local network/VPN the risks seem minimal?)

**Comment by pavlobu** [comment]
by @fbuenonet

> Although I do not think it is a bug, if I want to comment here on something that I have noticed happens. I am always connected to the Internet through a VPN and when Deskreen takes the IP to compose the connection url, it takes it from the virtual connection instead of taking it from the main connection, either eth or wlan.
> 
> The two devices are not always connected to the VPN, although they are to the same LAN, so I think the IP should be taken from the main network adapter.
> 
> I think it is an excellent job and I want to thank you for your effort to offer such a useful tool as Deskreen. I can only say one more thing: Raspberry Pi ;-)

Hi! thank you for liking it, Enjoy using Deskreen!

Sure I'm prioritizing to make this feature in newer releases. Stay tuned for updates. And pls don't open new discussion for such occasions, you can use existing ones. I removed your created discussion and reposted here.

Thanks!

**Comment by tigerinus** [comment]
WinRT support please.

I've got an old Surface RT covered by dirt. It's working except there is not much app. However it does support 5Ghz wifi. It'd be super helpful to turn it into a secondary monitor. Thanks!

**Reply by aaylnx** [reply]
> WinRT support please.
> 
> I've got an old Surface RT covered by dirt. It's working except there is not much app. However it does support 5Ghz wifi. It'd be super helpful to turn it into a secondary monitor. Thanks!

If the Surface RT is the 2nd screen then it should already work since all the 2nd screen needs is a browser which supports WebRTC.

**Reply by tigerinus** [reply]
It's not chromium core though.

**Reply by pavlobu** [reply]
> It's not chromium core though.

You can install chrome or firefox on it an use as a secondary screen

**Reply by tigerinus** [reply]
Last time I checked you can't install anything on Surface RT except apps from Windows Store. There is no Chrome in Windows Store.

**Reply by aaylnx** [reply]
What about Edge? That is now Chromium.

**Comment by kirk86** [comment]
Hi folks,
I was wondering if there's any possibility to also use `Dskreen` to mirror my tablet/ipad screen onto my laptop?
I have an old ipad 2 2011 and would like to mirror its screen on my laptop during video-conferencing, is that possible?

**Reply by pavlobu** [reply]
Hi, you can't do that, here is the answer
https://deskreen.com/?#faq-28

**Reply by kirk86** [reply]
Thanks for the prompt reply! 😭

**Comment by chandrath** [comment]
First of all thanks for this amazing piece of software and for making the source open. I have just one question/feature request.

Can we have LAN to WIFI?

Suppose the First Device (PC) connected to the internet via LAN and the Secondary Device (Android Phone) which is connected to the Internet via WIFI (Both using the same modem/router), Can we able to beam the screen in this circumstance? or can we expect this feature coming 😅

**Comment by opinsky** [comment]
Thanks for the fantastic project. It would be great to have Deskreen work in **Windows on Arm**.  Even better if the app is compiled natively, not JIT translation. Unfortunately this is way beyond my coding level, so my contribution is limited to the request, unfortunately.

**Comment by meiser79** [comment]
Hi,

I'd like to see the possibility for reverse connections, meaning Deskreen sends the stream to a specified IP address (via RTMP?).
The reason behind it is that my company force-enables the firewall via MDM so that Deskreen is not reachable. If Deskreen initiated the connection, the traffic would be passed through the firewall (at least this is my assumption).

Thanks in advance,
meiser

**Comment by NicKoehler** [comment]
I'm trying to restyle the logo to make it look better, what you guys think? I'm trying to make it look modern and recognizable. Your suggestions are appreciated
![image](https://user-images.githubusercontent.com/53040044/122672948-7ddc7480-d1ce-11eb-9018-0ae8af7eb760.png)
![image](https://user-images.githubusercontent.com/53040044/122673431-b715e400-d1d0-11eb-8fd6-e7b3c69ccc77.png)

**Reply by pavlobu** [reply]
@NicKoehler thank you for your effort!
What I think about picture one, Deskreen is not only working through wifi, but also through local LAN and ethernet so it may be not telling the features completely

second picure, nice association with share icon, but the eventual logo looks like it has too much small details

last two pictures are looking like a general screen sharing icons, maybe we need something more peculiar

**Reply by NicKoehler** [reply]
@pavlobu thanks for your feedback. If you have ideas too, I can make it into logos and discuss it

**Reply by pavlobu** [reply]
sure. I actually came up with this logo myself and asked my friend to make it in photoshop.
here are my first sketches. The idea behind it is that the current logo is based on golden ratio:

![9A3B7034-DB81-4C4C-84C1-833B41065F23](https://user-images.githubusercontent.com/14937251/122821585-b58a1000-d2e5-11eb-8f9e-f2a999179b76.PNG)

**Comment by NicKoehler** [comment]
Meanwhile I came up with this, hope to hear some feedback to improve it (shape, font, colors, ecc)
![immagine](https://user-images.githubusercontent.com/53040044/122743833-ad02ec80-d287-11eb-8ca1-a0733b046052.png)

**Reply by pavlobu** [reply]
Great work, I appreciate your effort!
Let's wait if someone comes with some others designs so we can compare and make a right choice

**Reply by EDIflyer** [reply]
Ooh this looks really smart, I like it!

**Comment by shivam2562001** [comment]
Hi there,
I want to add this two features i.e. :
1. Remote control of computer
2. Stylus support (for drawing) for ipad

I am familiar with electron , react , react native. but I need some advice regarding the prerequisite for above mention features.
can anyone give me some link or tag resources to start with ? 
meanwhile I am going through repo and understanding code,where to add this feature and in which file I have to amend the changes.

**Comment by juju4** [comment]
Along raspberry support (https://github.com/pavlobu/deskreen/issues/59), pine64 (https://www.pine64.org/ pinephone, pinebook...) support and package releases would be great too. Arm based but 64bits
cross-compiling pointers
https://wiki.pine64.org/wiki/Cross-compiling

missing compiling instructions (no wiki or BUILD file/directory in docs dir)
tried native build
```
mobian@mobian:~$ npm install yarn cross-env react-scripts react-dom i18next
mobian@mobian:~$ cd Downloads/deskreen/
mobian@mobian:~/Downloads/deskreen$ yarn build-client
yarn run v1.22.11
warning ../../package.json: No license field
$ cd app/client && cross-env SKIP_PREFLIGHT_CHECK=true yarn build
warning ../../../../package.json: No license field
$ react-scripts build
error TS6046: Argument for '--jsx' option must be: 'preserve', 'react-native', 'react'.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by Fancy2209** [comment]
One nice thing would be an option to Minimize Deskreen to tray so you don't need to have the window open to have it running

**Comment by leuchtrakete666** [comment]
implement usb support. at least for android devices its so easy to do.

enable usb debugging on the android tablet and use ` adb reverse tcp:3131 tcp:3131` on the pc. than connect `http://127.0.0.1:3131/xxxxxxx`

i use it this way and deskreen becomes extremly responsive. almost zero latancy.

**Reply by rayman3003** [reply]
> 
> 
> implement usb support. at least for android devices its so easy to do.
> 
> enable usb debugging on the android tablet and use ` adb reverse tcp:3131 tcp:3131` on the pc. than connect `http://127.0.0.1:3131/xxxxxxx`
> 
> i use it this way and deskreen becomes extremly responsive. almost zero latancy.

Can u give me an easy description for how to use deskreen for android on windows (without tethering)?

**Reply by xros** [reply]
Yes how did you make it? I used the hotspot on the Android Pad connecting to my MacBook Pro, maximumly the Tx rate bandwidth is 72 mbps only at 2.4Ghz 20Mhz hotspot channel. So laggy. I need usb support so badly. Can you explain more in details? Thx.

**Reply by xros** [reply]
I tried. And it works on my M1 MacBook Pro. Maybe it's because my Android Pad is still using USB 2.0, I feel so much lag..
And image quality is not good.

**Reply by WLaoDuo** [reply]
may i can explain how to do it:
1. you must have **"adb"** tools. (if you don't know ,plz google it).AND open **"deskreen.exe"**
2. Use USB cable to connect the mobile phone to the computer. 
3. command **"adb devices"** can see whether the mobile phone is connected successfully.If there is a device,your phone is online,you can continue
![图片](https://user-images.githubusercontent.com/69497874/166402195-4f1b0700-e14f-4130-97b3-51eac10ba18e.png)
4.command **"adb reverse tcp:3131 tcp:3131"**
5.using your moblie phone,open browser,visit **"http://127.0.0.1:3131/xxxxxx"**
![1](https://user-images.githubusercontent.com/69497874/166402518-a6fdff7c-9922-464a-a300-1dd8565cbfba.png)

**Comment by takase1121** [comment]
I'm not sure what Deskreen does, but my screen always stutter when it's on mobile. It doesn't do that on PC, which means my phone is a potato and cannot handle this. Yet I get acceptable framerate with no stutters with Google remote desktop, so is there anything I can do to fix this?

**Comment by vwidmer** [comment]
It would be nice if can start Deskreen via terminal with some commands like app to mirror or screen to mirror as well be able to get the url back. So can use it in scripts. Thanks

**Reply by EDIflyer** [reply]
Fantastic idea - would be great for remote access to a desktop over VPN.

**Reply by klausfmh** [reply]
Hey, Deskscreen is a small miracle. Just discovered it yesterday and now I have two external screens on a laptop that has only one physical HDMI connector (raspberry is driving the 'remote screen'. Lovely.

I'd be really interested in a scripted / automated approach that would allow me to 'just startup with machines'  so a command line interface or a web API (with some credentials) would be perfect and allow me to script Deskreen.

**Comment by 363843342** [comment]
I hope it can support GPU hardware encoding in the future(including h265&av1), which can effectively reduce CPU usage.

**Comment by benz145** [comment]
Feature request (or maybe this is already possible?): Ideally I would like to actually be able to control the computer on which Deskreen is installed via the remote device. This would be incredibly useful because it would mean you could beam individual applications from one device to another. Specifically this would allow one to use Mac-exclusive applications (ie: iMessage) on a Windows computer, or vice-versa.

**Comment by skunkmommy** [comment]
The requirement to type in a URL every time I connect is annoying. QR code support makes that easier on my phone, but not all devices have a camera. (My Steam Deck for example.)
It's kinda fiddly to have to type in an access code every time. I'd rather it worked like Bluetooth where as soon as you open the client page, the server directly asks if you wanna connect to that client.

**Comment by AutonomousCat** [comment]
`adb reverse tcp:3131 tcp:3131` works for a usb only connection, but without it being built in, it's considerably more difficult (QR code is wrong, need to open up terminal and manually enter the command, etc) Makes me want to look for another app that already supports usb.

**Comment by popsc30** [comment]
Request a  apple silicon version, thanks

**Comment by denissyk2004** [comment]
Can I get a safe mode? I am a teacher and need my students to see my screen in the classroom. It must connect new screens one by one. That will be great if there is a switch let the IP is fixed and open for everyone who knows the link.

**Reply by pavlobu** [reply]
this feature is available now in Deskreen Teams version

**Comment by JoshAudette** [comment]
Hi, just found Deskreen and I'm loving it!  Much faster / better framerate than VNC.

BTW, for virtual display, I am using the `evdi` kernel module (modprobe evdi) which gives you dummy virtual displays.

One issue I have is that with VPN on, I get this warning "No WiFi and LAN connection".  It would be wonderful if the detection for this could be disabled, e.g. in an "Advanced" menu or from CLI parameter e.g. `deskreen -no-wifi-warn` or something.

As well, I am using USB tethering to use my tablet as second display.  So I have a network interface for WiFi device, a network interface for USB device, and a network interface for VPN.  So, it would be great if there was an option to let the user choose which network device that Deskreen will listen on.  

e.g. `deskreen -dev wlp1s0` or `deskreen -dev enx0` or whatever.

Finally, just want to give a big thanks for your awesome project!  And hugs and best wishes for the welfare of Ukraine.

**Reply by james-yakura-2** [reply]
Or, better yet: Deskreen can already maintain connections on different interfaces (I just tested it with WiFi and Ethernet simultaneously), so having it generate addresses for each available network device rather than having to disable its preferred device to get a less-preferred device.

**Reply by james-yakura-2** [reply]
Update: This is already doable! Deskreen listens to both WiFi and Ethernet for connection requests. Just use http://[local address of source on desired network]:3131/[six digits], regardless of what address the source is showing you.

**Comment by alex-larkin** [comment]
# I request a simplified mode of Deskreen Pro
Hi there! 

Assistant to a Deskreen Pro user here. While some may have the use case of wanting to verify each connecting device before sharing the screen, that requirement creates a lot of delay and interruptions when many devices want to view the host's screen (and sometimes reconnect in the middle of a presentation or work session). Many users will just want to automatically grant access to anyone on the same network that has the Deskreen URL, without getting any more prompts. I would like to request that a simplified mode be enabled:
* Don't require each device to connect to a unique url (no unique numerical code at the end)
* Don't prompt to name each device
* Don't ask what to share with each device. Set just once and apply selection to all connecting devices
* Allow device browsers to save http://deskreen.local (or the IP address given by Deskreen host) as a favorite in the device browser for easy access

I understand the current flow exists for security reasons, so the simplified mode could be an opt-in mode for users on trusted private networks.

Would this be possible?

Thanks so much.

**Reply by pavlobu** [reply]
Hi, this feature exists now in Deskreen Teams version. It is created for extended personal use or for use in classroom-like environments.

**Reply by alex-larkin** [reply]
Hi there! Thanks for the quick response. I also just realized that this is indeed supported with the "Public sharing" (local network) option, but I seem to be running in to a bug. I will submit a bug report. Thanks!
